From a5ab9aaa53acac4c4155b6c4cc8a86bc59278e53 Mon Sep 17 00:00:00 2001
From: Quirin Gylstorff <quirin.gylstorff@siemens.com>
Date: Thu, 15 Jun 2023 18:48:25 +0200
Subject: [PATCH 2/4] d/rules: Add seperate build_profile option for delta
 update

Delta update need zchunk 1.3 to work on memory restricted
devices. zchunk 1.3 is is not available in bullseye-backports
and Debian bookworm(12).

Signed-off-by: Quirin Gylstorff <quirin.gylstorff@siemens.com>
---
 debian/control | 2 +-
 debian/rules   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/debian/control b/debian/control
index e47c320c..d4b17972 100644
--- a/debian/control
+++ b/debian/control
@@ -26,7 +26,7 @@ Build-Depends: debhelper-compat (= 13),
                uuid-dev,
                zlib1g-dev,
                libzstd-dev,
-               libzck-dev (>= 1.2) <!pkg.swupdate.bpo>,
+               libzck-dev (>= 1.3) <pkg.swupdate.delta>,
                libp11-kit-dev <pkg.swupdate.p11>,
                libwolfssl-dev <pkg.swupdate.p11>,
                libmtd-dev [linux-any],
diff --git a/debian/rules b/debian/rules
index 91106fbd..8cac6359 100755
--- a/debian/rules
+++ b/debian/rules
@@ -38,7 +38,7 @@ ifeq (linux,$(DEB_HOST_ARCH_OS))
 else
 	echo "# CONFIG_MTD is not set"     >> configs/debian_defconfig
 endif
-ifeq (,$(filter pkg.swupdate.bpo,$(DEB_BUILD_PROFILES)))
+ifneq (,$(filter pkg.swupdate.delta,$(DEB_BUILD_PROFILES)))
 	echo CONFIG_DELTA=y >> configs/debian_defconfig
 endif
 ifneq (,$(findstring $(DEB_HOST_ARCH),amd64 i386 arm64 armhf riscv64 ia64))
-- 
2.39.2

